### DISCUSSION TITLE
Negative number in pattern matching

### DISCUSSION THREAD

**Discussion by CreateAndInject** [dotnet/roslyn#65337] [General]
```C#
        object obj = null;
        if (obj is short x && x == -1) // How to simplify by pattern matching
            Console.WriteLine();
```

Should be `if (obj is short and -1)`?
It's invalid, but why `if (obj is short and 1)` is valid?

**Comment by RikkiGibson** [comment]
Change it to `if (obj is short and (-1))` to work around it. @CyrusNajmabadi is this a parsing bug in Roslyn?

**Comment by 333fred** [comment]
https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYAMBYAUBgRjzwHsAjAKwFMBjYAAnIvoF56oIAbTgbjwEsAZvQAUTevwDO9SQAsSYBgA96AQyhx6AWgIBKegHoD9ABIkA7vWAkZ/ALYAHTkICe9Mm4ergwKmCj0dt40svxQAOZ49NH0qAQAnCK6PEA= seems to work for me, and it also compiles on .NET 7. Can you specify where you're seeing a problem?

**Reply by 333fred** [reply]
Alright, further discussion with @RikkiGibson showed the issue: we're parsing the `and` as a variable name. Moving this to roslyn.

**Comment by CreateAndInject** [comment]
> https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYAMBYAUBgRjzwHsAjAKwFMBjYAAnIvoF56oIAbTgbjwEsAZvQAUTevwDO9SQAsSYBgA96AQyhx6AWgIBKegHoD9ABIkA7vWAkZ/ALYAHTkICe9Mm4ergwKmCj0dt40svxQAOZ49NH0qAQAnCK6PEA= seems to work for me, and it also compiles on .NET 7. Can you specify where you're seeing a problem?

Since you declare `x`, remove it to reproduce.
